### PR TITLE
Add a pydantic model which encapsulates required BMI time meta data

### PR DESCRIPTION
**Pull request by hellkite500**
This model can be used in BMI implementations to store related time components and validate the BMI constraints of these meta data.

### CODE REVIEW

File: bmi_sdk/tests/test_bmi_time.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from ..bmi_time import BmiTime
+from pydantic import ValidationError
+import pytest
+
+@pytest.mark.parametrize('unit', ["s", "sec", "second", "seconds",
```
Comment (aaraney) [inline bmi_sdk/tests/test_bmi_time.py:5]: The interchange between single and double quotes is going to make my hair grey faster hahah, but I figured you might do that on purpose lol. On a more serious note, what are your thoughts on using something like `black` in CI to enforce a certain formatting style? I could get it set up with pre-commit easily so you really don't have to think about it. Just nice to have things auto-formatted for you even if you don't always like exactly how it formats things.

**Review by aaraney — CHANGES_REQUESTED** [review]
Overall, looks good! I think we can improve on the one `pydantic` model by using `typing.Literal`s instead of a custom validator. LMK if I wasn't clear about that or you would like to provide an example.

File: bmi_sdk/tests/test_bmi_time.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from ..bmi_time import BmiTime
```
[the remark this answers is not in the document]
Comment (hellkite500) [reply bmi_sdk/tests/test_bmi_time.py:1]: the tests run and pass 🤷

File: bmi_sdk/tests/test_bmi_time.py (hunk 1 above)
Comment (hellkite500) [reply bmi_sdk/tests/test_bmi_time.py:5]: I'm 💯 ok with that haha.  Formatting can hard to get correct when the focus in on semantics.

Comment (aaraney) [reply bmi_sdk/tests/test_bmi_time.py:5]: Sweet, just opened up #24 to add this. Ill add it to CI once this has merged.

**Comment by aaraney** [comment]
@hellkite500, just as a heads up, I changed `valid_units` to a `Literal` and for better or worse formatted the codebase using `black`. It is at least split up in the history.

**Review by taddyb — APPROVED** [review]
LGTM 👍 Nice addition with the Literals

File: bmi_sdk/bmi_sdk/bmi_time.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "min",
+    "minute",
+    "minutes" "h",
+    "hr",
+    "hour",
+    "hours",
+    "d",
+    "day",
+    "days",
+]
```
Comment (hellkite500) [inline bmi_sdk/bmi_sdk/bmi_time.py:19]:
```suggestion
TimeUnits = Literal[
    "s",
    "sec",
    "second",
    "seconds",
    "min",
    "minute",
    "minutes",
    "h",
    "hr",
    "hour",
    "hours",
    "d",
    "day",
    "days",
]
```

File: bmi_sdk/tests/test_bmi_time.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        "min",
+        "minute",
+        "minutes" "h",
+        "hr",
+        "hour",
+        "hours",
+        "d",
+        "day",
+        "days",
+    ],
```
Comment (hellkite500) [inline bmi_sdk/tests/test_bmi_time.py:23]:
```suggestion
    "unit",
    [
        "s",
        "sec",
        "second",
        "seconds",
        "min",
        "minute",
        "minutes",
        "h",
        "hr",
        "hour",
        "hours",
        "d",
        "day",
        "days",
    ],
```

Comment (aaraney) [reply bmi_sdk/tests/test_bmi_time.py:23]: Hmmm, yeah it looks like there was a missing `,` in the original code before I formatted it. Great catch!

File: bmi_sdk/bmi_sdk/bmi_time.py (hunk 3 above)
Comment (hellkite500) [reply bmi_sdk/bmi_sdk/bmi_time.py:19]: @aaraney I think you missed this one

**Pull request merged**